clarity/src/vm/functions/database.rs

```diff
@@ -924,3 +947,188 @@ pub fn special_get_burn_block_info(
+        CheckErrors::NoSuchStacksBlockInfoProperty(property_name.to_string()),
+    )?;
+
+    // Handle the block-height input arg.
+    let height_eval = eval(&args[1], env, context)?;
+    let height_value = match height_eval {
+        Value::UInt(result) => Ok(result),
+        x => Err(CheckErrors::TypeValueError(TypeSignature::UIntType, x)),
+    }?;
+
+    let Ok(height_value) = u32::try_from(height_value) else {
+        return Ok(Value::none());
+    };
+
+    let current_block_height = env.global_context.database.get_current_block_height();
+    if height_value >= current_block_height {
+        return Ok(Value::none());
+    }
+
+    let result = match block_info_prop {
+        StacksBlockInfoProperty::Time => {
+            let block_time = env.global_context.database.get_block_time(height_value)?;
+            Value::UInt(u128::from(block_time))
+        }
+        StacksBlockInfoProperty::HeaderHash => {
+            let header_hash = env
+                .global_context
+                .database
+                .get_block_header_hash(height_value)?;
+            Value::Sequence(SequenceData::Buffer(BuffData {
+                data: header_hash.as_bytes().to_vec(),
+            }))
+        }
+        StacksBlockInfoProperty::IdentityHeaderHash => {
+            let id_header_hash = env
+                .global_context
+                .database
+                .get_index_block_header_hash(height_value)?;
+            Value::Sequence(SequenceData::Buffer(BuffData {
+                data: id_header_hash.as_bytes().to_vec(),
+            }))
+        }
+    };
+
+    Value::some(result)
+}
+
+/// Handles the function `get-tenure-info?` special function.
+/// Interprets `args` as variables `[property-name, block-height]`, and returns
+/// a property value determined by `property-name`:
+/// - `time` returns the burn block time for the tenure of which `block-height` is a part
+/// - `vrf-seed` returns the VRF seed for the tenure of which `block-height` is a part
+/// - `burnchain-header-hash` returns header hash of the burnchain block corresponding to the tenure of which `block-height` is a part
+/// - `miner-address` returns the address of the principal that mined the tenure of which `block-height` is a part
+/// - `miner-spend-winner` returns the number of satoshis spent by the winning miner for the tenure of which `block-height` is a part
+/// - `miner-spend-total` returns the total number of satoshis spent by all miners for the tenure of which `block-height` is a part
+/// - `block-reward` returns the block reward for the tenure of which `block-height` is a part
+///
+/// # Errors:
+/// - CheckErrors::IncorrectArgumentCount if there aren't 2 arguments.
+/// - CheckErrors::GetTenureInfoExpectPropertyName if `args[0]` isn't a ClarityName.
+/// - CheckErrors::NoSuchTenureInfoProperty if `args[0]` isn't a TenureInfoProperty.
+/// - CheckErrors::TypeValueError if `args[1]` isn't a `uint`.
+pub fn special_get_tenure_info(
```

Comment:
I'm confused by the input domain here.  Is it supposed to take:

* a Bitcoin block height?
* a sortition height (e.g. Bitcoin block height - 666050)?
* a Stacks block height?

Asking because a lot of these fields are only meaningful in the domain of either Bitcoin or sortition heights; not so much a Stacks height (e.g. what is the `miner-spend-total` for a non-tenure-change block?)

Reply:
The input is a Stacks block height, but the info is relevant to the tenure in which that block resides.